src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -141,6 +141,12 @@ TryToDelegateFunctionCall(Query *query, bool *hasExternParam)
 		return NULL;
 	}
 
+	if (query->commandType != CMD_SELECT)
```

Comment:
Well, I'm also good with removing the above check. `if (query == NULL)` we'd have already crashed :)